### ISSUE TITLE
`--maxr` ignored

### ISSUE THREAD

**Issue by KEggensperger** [automl/SMAC3#676]
#### Description
Use SMAC's CLI interface with arguments shown by `--help-all`

#### Steps/Code to Reproduce
```
cd examples/spear_qcp
python3 ../../scripts/smac --scenario scenario.txt --verbose DEBUG --mode SMAC4AC --maxr 10
```
#### Expected Results
Maxinum number of calls per configuration is 10.

#### Actual Results
SMAC calls a configuration at most 2000 times, the argument is ignored and set to its default value.

Can also be verified with adding
```
print("maxR", maxR)
sys.exit(1)
```
here: https://github.com/automl/SMAC3/blob/master/smac/intensification/intensification.py#L107

#### Versions

0.12.3

**Comment by Mustrumion** [comment]
Likely related to this topic: 
I also encountered this issue while using Python `SAMCHPO` and `SMACAC` facades.
I took a look into the code and `maxR` and few other params defined in the scenario are not passed forward if the Intensifier parameter is None, which is the default (the exact condition is that it must be or inherit Intensifier).
There is an easy workaround - if the `maxR` is defined in `intensifier_kwargs` instead of scenario, it gets properly passed to the Intensifier.
